2024-08-04-python-pattern-matching.md

```diff
@@ -0,0 +1,139 @@
+# Pattern Matching
+
+Pattern matching is a new feature in Python 3.10
+https://peps.python.org/pep-0636/
+
+Example:
+
+```python
+def length(items):
+    match items:
+        case []: return 0
+        case [x, *xs]: return 1 + length(xs)
+```
+
+I don't have a specific use case for pattern matching in Copr, this is just a
+general research document.
+
+
+## Exhaustive Pattern Matching
+
+### Prerequisites
+
+Create a `pyproject.toml` file with this minimal configuration
+
+```
+[tool.pyright]
+typeCheckingMode = "strict"
+```
+
+Install `pyright`. You will have to use `pip` because it is not available in
+Fedora.
+
+```
+pip install --user pyright
```

Comment:
how pyright relates to pattern matching?

Reply:
It can check for exhaustiveness, see bellow:

```
Unhandled type: "Literal[Color.BLUE]"
```